pkg/controllers/node/initialization.go

```diff
@@ -73,32 +73,41 @@ func (r *Initialization) getInstanceType(ctx context.Context, provisioner *v1alp
 // This method handles both nil provisioners and nodes without extended resources gracefully.
 func (r *Initialization) isInitialized(n *v1.Node, provisioner *v1alpha5.Provisioner, instanceType cloudprovider.InstanceType) bool {
 	// fast checks first
-	return node.GetCondition(n, v1.NodeReady).Status == v1.ConditionTrue &&
-		isStartupTaintRemoved(n, provisioner) && isExtendedResourceRegistered(n, instanceType)
+	if node.GetCondition(n, v1.NodeReady).Status != v1.ConditionTrue {
+		return false
+	}
+	if _, ok := IsStartupTaintRemoved(n, provisioner); !ok {
+		return false
+	}
+
+	if _, ok := IsExtendedResourceRegistered(n, instanceType); !ok {
+		return false
+	}
+	return true
 }
 
-// isStartupTaintRemoved returns true if there are no startup taints registered for the provisioner, or if all startup
+// IsStartupTaintRemoved returns true if there are no startup taints registered for the provisioner, or if all startup
 // taints have been removed from the node
-func isStartupTaintRemoved(node *v1.Node, provisioner *v1alpha5.Provisioner) bool {
+func IsStartupTaintRemoved(node *v1.Node, provisioner *v1alpha5.Provisioner) (*v1.Taint, bool) {
```

Comment:
We should probably return all startupTaints that are causing this to return false